modules/customersUI.js

```diff
@@ -0,0 +1,223 @@
+import { expect } from "@playwright/test";
+import { VALID_USER_CREDENTIALS } from "../fixtures/credentials.js";
+import { URLS } from "../fixtures/urls.js";
+import { PRODUCT_IMAGE } from "../fixtures/images.js";
+
+export class CustomersUI {
+  constructor(page) {
+    this.page = page;
+    this.product = page.locator("div[test-data='product-container']");
+    this.productName = this.product.locator("h1");
+    this.productImage = this.product.locator("img");
+    this.addProductToCartButton = this.product.locator("button");
+    this.firstProduct = this.product.nth(0);
+    this.headerAfterLogin = page.locator(".max-w-full");
+    this.button = this.headerAfterLogin.locator("button");
+    this.cartButton = this.button.nth(0);
+    this.logoutButton = this.button.nth(1);
+    this.cartUpperWindow = page.locator("section").nth(0);
+    this.clearAllItemsFromCartButton = this.cartUpperWindow.locator("button");
+    this.cartWindowContents = page.locator("section").nth(1);
+    this.itemInCart = this.cartWindowContents.locator("div");
+    this.removeItemButton = this.itemInCart.locator("button").nth(0);
+    this.filters = page.locator(".p-ripple");
+    this.filterMenu = page.locator("ul").first();
+    this.selectFiltersButton = this.filterMenu.locator("button").first();
+    this.disselectFiltersButton = this.filterMenu.locator("button").last();
+    this.searchBar = page.locator("#search");
+    this.spinner = page.locator(".h-screen > .absolute > .h-48").last();
+    this.pageSwitchDiv = page.locator(".paginated");
+    this.pageSwitchButton = this.pageSwitchDiv.locator("button");
+    this.priceSliderStart = page.locator(".p-slider-handle-start");
+    this.priceSliderEnd = page.locator(".p-slider-handle-end");
+    this.priceSliderTrack = page.locator(".p-slider-range");
+    this.priceFilterLi = page.locator(".pt-4 > div").nth(1);
+    this.priceText = this.priceFilterLi.locator("span").nth(1);
+  }
+
+  async addProductToCart({
+    valid = true,
+    userID = VALID_USER_CREDENTIALS["VALID_ID"],
+    productID = 1,
+  }) {
+    await this.page.goto(URLS["DASHBOARD"]);
+    await expect(this.page).toHaveURL(URLS["DASHBOARD"]);
+    await expect(this.product).toHaveCount(24);
```

Comment:
zasto je page visit deo fukcije koja se zove add to cart? nema logike u ovome

Takodje proveravas da postoji tacno 24 produkta, sta ako se promeni koliko ih ima na stranici... to vise treba da bude test koji proverava paginaciju, nego ovo...